File: frontend/src/types/Category.ts
export type CategoryColour = 'red' | 'yellow' | 'green' | 'blue' | 'indigo' | 'purple' | 'pink'

export type Category = {
	id: number
	title: string
	colour: CategoryColour
	created_at: string
	updated_at: string
}

File: backend/app/models/category.rb
class Category < ApplicationRecord
  validates :title, presence: true
  validates :colour, presence: true,
                     inclusion: {
                       in: %w[red yellow green blue indigo purple pink],
                       message:
                          'Please pick a colour from red, yellow, green, blue, indigo, purple or pink'
                     }

  has_many :articles
end
